### DISCUSSION TITLE
Double layered Text is not redacted

### DISCUSSION THREAD

**Discussion by MallRoy** [pymupdf/PyMuPDF#1220] [Q&A] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
I have a document with double-layered text. One Layer of the text is redacted and another one is left as it is. Both the text are in the same area.

## To Reproduce (mandatory)
[microsoftout.pdf](https://github.com/pymupdf/PyMuPDF/files/7019136/microsoftout.pdf)
```
# imports
import fitz
import sys
from matplotlib import colors


class Redactor:
        doc = fitz.open('microsoftout.pdf')
        # iterating through pages
        for page in doc:
            if not page._isWrapped:
                page.wrapContents()

            currentpage = page.number
            dl = page.getDisplayList()
            tp = dl.getTextPage()
            clips=tp.search('cloud',quads=False)
            for clip in clips:
                print(clip)
                getInfo=fitz.Rect(clip)
                print(getInfo)
                delta = fitz.Point(0, getInfo.height) * 0.3
                new_rect = fitz.Rect(getInfo.tl + delta, getInfo.br - delta)
                page.addRedactAnnot(new_rect, fill=colors.to_rgb('yellow'))
                page.apply_redactions()

        # saving it to a new pdf
        doc.save('out.pdf',garbage=4,deflate=True,clean=True,linear=True)
        print("Completed")

# driver code for testing
if __name__ == "__main__":

    lines = sys.stdin.readlines()
    redactor = Redactor()
    redactor.redaction()
```
## Expected behavior (optional)
Both the double-layered text should be removed.

## Screenshots (optional)
![image](https://user-images.githubusercontent.com/54269166/130179572-6a5ada94-c67e-4036-8509-c1242441f451.png)

## Your configuration (mandatory)
- Windows 10

**Comment by JorjMcKie** [comment]
The text has been removed correctly. This:
![grafik](https://user-images.githubusercontent.com/8290722/130192919-e9d77b2d-06d5-48d8-89b4-58bda672739a.png)
is no text, but drawings put underneath the original text.

**Comment by MallRoy** [comment]
is there any way to remove the drawings?

**Comment by JorjMcKie** [comment]
no, only text, images, links
all you can do is putting a filled rectangle over drawings, e.g. fill=[1] parameter in the redact annot

**Comment by JorjMcKie** [comment]
in your case, use `fitz.TOOLS.set_small_glyph_heights(True)` then search
this, in this doc allows removing text without touching lines above / below and provides exactly fitting rects for also covering any stuff underneath

**Comment by JorjMcKie** [comment, marked as answer]
you can of course check `page.get_drawings()` to see if any of those overlaps redact annots

**Comment by JorjMcKie** [comment]
another tip: it is sufficient to apply redactions once per page, not after each single add of a redact annot.

**Comment by MallRoy** [comment]
I tried setting globally, but it doesn't help me.
fitz.TOOLS.set_small_glyph_heights(True)

**Comment by JorjMcKie** [comment]
> I tried setting globally, but it doesn't help me.

what happened - it did when I tested your case.

**Reply by MallRoy** [reply]
I am not using the coordinates of mupdf. If I use the mupdf coordinates it works perfectly.
